### PR TITLE
[release-5.19 backport] Cirrus: Update to F36 VMs

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Chris Evich <cevich@redhat.com>
(cherry picked from commit 90eec841dfd667eaec52ab1ec1e5d8aef8f570e5)
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

### CODE REVIEW

**Comment by lsm5** [comment]
```
CGO_CFLAGS="" CGO_LDFLAGS="-L/usr/lib64 -lgpgme" GO111MODULE=on go build -mod=vendor "-buildmode=pie" -ldflags '-X main.gitCommit=1fe8da63a9ed49de41e8c185e6c9970fafdb13dc ' -gcflags "" -tags "btrfs_noversion libdm_no_deferred_remove" -o bin/skopeo ./cmd/skopeo
# github.com/containers/skopeo/cmd/skopeo
cmd/skopeo/copy.go:249:4: unknown field 'SignIdentity' in struct literal of type copy.Options
make: *** [Makefile:140: bin/skopeo] Error 2

Exit status: 2
```

does this need a change in the skopeo version used in cirrus?

**Comment by mtrmac** [comment]
Yes — update `SKOPEO_BRANCH`.

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ export GOPROXY=https://proxy.golang.org
 
 # Which github repository and branch to use for testing with skopeo
 SKOPEO_REPO = containers/skopeo
-SKOPEO_BRANCH ?= master
+SKOPEO_BRANCH ?= main
```
Comment (mtrmac) [inline Makefile:7]: Uh, that was apparently left over — I guess we were saved by a redirect.

But in this case the 5.19 version of c/image needs to be testing with a 5.19-using version of Skopeo, something like `v1.6.2.

Comment (lsm5) [reply Makefile:7]: hmm, so IIUC, that would be handled in `SKOPEO_CIDEV_CONTAINER_FQIN: "quay.io/libpod/skopeo_cidev:${IMAGE_SUFFIX}"` ?? I see that the IMAGE_SUFFIX is a quay.io repo tag and I don't think there's an easy mapping of repo tag <-> skopeo version, or am I looking in the wrong place?

/cc @cevich

Comment (mtrmac) [reply Makefile:7]: Oops, my mistake, `SKOPEO_REPO` and `SKOPEO_BRANCH` seems to be completely unused now.

The right variable is `SKOPEO_CI_TAG` in `.cirrus.yml`; that’s consumed in `contrib/cirrus/runner.sh` to check out the relevant Skopeo version.

Comment (mtrmac) [reply Makefile:7]: Those unused variables will be fixed on the main branch by https://github.com/containers/image/pull/1541.

**Comment by mtrmac** [comment]
> sed: can't read /etc/containers/storage.conf: No such file or directory

… So Skopeo will need (some parts of) https://github.com/containers/skopeo/pull , backported to the relevant branch.

The perils of being the first change on an older branch that is not being regularly tested…

**Comment by mtrmac** [comment]
@lsm5 https://github.com/containers/skopeo/pull/1637 should help with the `storage.conf` failure; then _this_ PR should change again, to target Skopeo’s (unreleased) `release-1.6` branch, rather than the specific `v1.6.2` tag.

File: Makefile (hunk 1 above)
Comment (cevich) [reply Makefile:7]: Ya, I think you're on the right track.  There is no hard correlation between branch, tag, and $IMAGE_SUFFIX.  So long as the `$IMAGE_SUFFIX` is consistent between VMs and container images, as was tested when the `$IMAGE_SUFFIX`.  The tricky thing for release branches is, we cannot easily go back and build new "old" images (for example if they were to get pruned due to disuse).  Hopefully that makes sense.

**Comment by cevich** [comment]
> The perils of being the first change on an older branch that is not being regularly tested…

Yep, you said it correctly :disappointed:   I'm seriously thinking this may need to change.  In recent memory, I think this is the third or fourth time there's been a problem(?).  While I can't do anything about old images that were already pruned. If there are release branches we should preserve going-forward (i.e. with "not yet dead" VM images), we can do something about that.  Just send me the branch names and/or [add them into the cirrus-cron schedule yourself (config page)](https://cirrus-ci.com/settings/repository/6429752730058752), the [docs are short and helpful too](https://cirrus-ci.org/guide/writing-tasks/#cron-builds).

**Comment by mtrmac** [comment]
> While I can't do anything about old images that were already pruned. If there are release branches we should preserve going-forward (i.e. with "not yet dead" VM images), we can do something about that.

If you asked _me_ yesterday, I wouldn’t have, from memory, pointed at this one as necessary…

Ideally I’d dream of RHEL needs driving this, so that every time we make a release that could plausibly need back ported fixes, we immediately branch _and_ mark the CI infrastructure for preservation (and periodic CI validation tests?). Assuming we want to pay for so much infrastructure, that is — IIRC that was a concern at some point.

**Comment by cevich** [comment]
> and periodic CI validation tests?

Technically the tests don't really even need to run.  It's that "meta" task that's the important thing.  At least that needs to run probably weekly, and be monitored (so we know if it stops running).  I'm not worried about the infrastructure or cost (pennies) at all if it's important/critical to RHEL.  It is an extra maintenance burden, but sometimes working CI for backports is necessary and  important.  I'm simply not qualified to be the judge of that.

Adding the e-mail monitoring is a pain, but it's a one-time up-front time-cost.
Adding cron-jobs for new branches is trivial.  It's just an entry in that web-form I linked.

**Comment by lsm5** [comment]
@mtrmac PTAL.

**Comment by lsm5** [comment]
ugh ..i thought i saw all green, nvm, looking ..

**Comment by mtrmac** [comment]
@lsm5 The keyring thing is probably fixed by backporting https://github.com/containers/image/pull/1477 . The other failure looks like an infrastructure flake.

**Comment by lsm5** [comment]
Closing this in favour of https://github.com/containers/image/pull/1542

**Pull request closed without merging**